client/src/partials/receipts/receipt.sale.js

```diff
@@ -67,6 +68,22 @@ angular.module('bhima.controllers')
       }
     };
 
+    function updateCurrency (currency) {
+      model.total.localeCost = model.common.doConvert(model.saleRecords[0].cost, currency, model.saleRecords[0].invoice_date);
```

Comment:
You should really cache this in a local variable.  Like this:
```javascript
var totals = model.total.localCost;
// do something fancy
```
This is both semantic and faster.  The performance increases [are](http://stackoverflow.com/questions/2936807/javascript-object-access-performance) [well](http://moduscreate.com/javascript-performance-tips-tricks/) [documented](http://www.javascriptkit.com/javatutors/efficientjs.shtml).